src/expr/e_parser.ml

```diff
@@ -594,6 +672,175 @@ and boundeds b = lazy begin
+                    introduced.
+
+                    For example, the tuple declaration `<<x, y>> \in S`
+                    would here result in the creation of two definitions:
+
+                        x == fcnbnd#xy[1]
+                        y == fcnbnd#xy[2]
+                    *)
+                    let letin =
+                        (* create one definition for each identifier that
+                        appears inside the tuple declaration
+                        *)
+                        List.mapi begin
+                        fun i op ->  (* arguments:
+                            - `i` is the 0-based index of the tuple element
+                            - `op` is the tuple element (an identifier)
+                            *)
+                            let e =
+                                (* tuple identifier, for example "fcnbnd#xy" *)
+                                let f = noprops (Opaque v.core) in
+                                (* 1-based index numeral *)
+                                let idx =
+                                    let i_str = string_of_int (i + 1) in
+                                    let num = Num (i_str, "") in
+                                    noprops num in
+                                (* function application on the index,
+                                for example:  fcnbnd#xy[1]
+                                *)
+                                let e_ = FcnApp (f, [idx]) in
+                                noprops e_ in
+                            (* definition for `op`, for example:
+
+                            x == fcnbnd#xy[1]
+
+                            The result is of type `defn`.
+                            *)
+                            let defn_ = Operator (op, e) in
+                            noprops defn_
+                        end vs in
+                    (* Bundle the constant declarations of fresh
+                    identifiers (in `bounds`) and the definitions (in terms of
+                    these fresh identifiers) of the identifiers that appeared
+                    in the tuple declaration (these definitions are in `letin`).
+
+                    These definitions are used at the call site to construct
+                    a new `LET...IN` expression that wraps the function's
+                    value expression (the `e` in `[... |-> e]`).
+
+                    The declarations are used, together with this `LET...IN`
+                    expression, to populate a function constructor `Fcn`.
+                    *)
+                    (bounds, letin)
+            end
+        ])
+    <$> begin
+      fun bss ->
+        (* Unzip the two lists.
+        `bss` is a list of pairs of lists, so `bounds` is a list of lists
+        and so is `letin`.
+        *)
+        let (bounds, letin) = List.split bss in
+        (* Flatten each list of lists into a list.
+        At this point we return a list of bounds that
```

Comment:
This sentence is unfinished.

Reply:
Thank you for noting this, I completed the comment.